lib/import/colorectal/providers/leeds/leeds_handler_colorectal.rb

```diff
@@ -1,696 +1,441 @@
+          def populate_variables(record)
+            @geno = record.raw_fields['genotype']&.downcase
+            @report = cleanup_report(record.raw_fields['report'])
+            @moleculartestingtype = record.raw_fields['moleculartestingtype']&.downcase
+            @indicationcategory = record.raw_fields['indicationcategory']&.downcase
+            @genes_panel = []
+          end
+
+          # checks if file has colorectal tests and should be processed under this importer
+          def should_process
+            filename = File.basename(@batch.original_filename)
+            return true if filename.match?(/MMR/i)
+            return true if filename.match?(/other|familial/i) &&
+                           @indicationcategory == 'cancer' &&
+                           @moleculartestingtype == 'familial' &&
+                           (@geno&.match?(/APC|EPCAM|LYNCH|MUTYH|PMS2/i) ||
+                           @report&.match?(MMR_GENE_REGEX))
+
+            false
+          end
+
+          def populate_and_persist_genotype(record)
             genocolorectal = Import::Colorectal::Core::Genocolorectal.new(record)
             genocolorectal.add_passthrough_fields(record.mapped_fields,
                                                   record.raw_fields,
                                                   PASS_THROUGH_FIELDS,
                                                   FIELD_NAME_MAPPINGS)
-            add_scope_and_type_from_genotype(genocolorectal, record)
-            mtype = record.raw_fields['moleculartestingtype']&.downcase&.strip
-            genocolorectal.add_molecular_testing_type_strict(TEST_TYPE_MAP_COLO[mtype]) \
-                           unless mtype.nil?
-            genocolorectal.add_test_scope(TEST_SCOPE_MAP_COLO[mtype]) \
-                           unless mtype.nil?
-            add_positive_teststatus(genocolorectal, record)
-            failed_teststatus(genocolorectal, record)
-            add_benign_varclass(genocolorectal, record)
+            add_scope(genocolorectal, record)
+            add_molecular_testingtype(genocolorectal, record)
+            add_varclass
             add_organisationcode_testresult(genocolorectal)
-            if genocolorectal.attribute_map['genetictestscope'].nil?
-              genocolorectal.add_test_scope(:no_genetictestscope)
-            end
-            res = add_gene_from_report(genocolorectal, record) # Added by Francesco
+            res = process_variants_from_record(genocolorectal, record)
             res.map { |cur_genotype| @persister.integrate_and_store(cur_genotype) }
           end
-          # rubocop:enable Metrics/AbcSize
 
-          def add_organisationcode_testresult(genocolorectal)
-            genocolorectal.attribute_map['organisationcode_testresult'] = '699C0'
+          def process_variants_from_record(genocolorectal, record)
+            genotypes = []
+            allocate_genes
+            find_test_status(record)
```

Comment:
A small thing but naming methods consistently is really helpful regarding readability. `allocate_genes` and `find_test_status` do similar things, i.e. setting an instance variable, so perheps calling it `allocate_test_status` might be more consistent?

Reply:
Thanks @ollietulloch , done in latest commit .